src/mgmt/cli_mgmt.c

```diff
@@ -407,13 +409,19 @@ int dc_mgmt_net_cfg(const char *name)
 	if (rc != 0)
 		D_GOTO(cleanup, rc = d_errno2der(errno));
 
+	sprintf(buf, "%d", info.provider_idx);
+	rc = setenv("CRT_SECONDARY_PROVIDER", buf, 1);
```

Comment:
CRT_SECONDARY_PROVIDER is always set no matter what value the provider_idx is? And I'm not quite sure how to determine the 'provider_idx' when primary and secondary are using same provider?

Reply:
I assume for primary provider the provider_idx will be 0? is that the case?
CART layer requires this to be set to non-0 for the clients running on the secondary cluster, and to 0/unset if running on the primary cluster

Reply:
Yes, it will be 0 if using the primary provider.

Reply:
And I'm not quite sure how to determine the 'provider_idx' when primary and secondary are using same provider?

Provider idx is tied to the order the parameters appear in the comma-separated provider string in the daos_server config file. So if you have a config: `provider: ofi+verbs,ofi+tcp`, idx 0 is ofi+verbs, idx 1 is ofi+tcp. If you wanted two TCP providers, you'd have: `provider: ofi+tcp,ofi+tcp`, so there would still be a provider at idx 1.

Reply:
shouldn't we set CRT_SECONDARY_PROVIDER to 0 when the provider_idx is 0?